### PR TITLE
Run the integration tests against a local fake server when the user p…

### PR DESCRIPTION
**Pull request by elopio**
…assword is not in the environment.

LP: #1585023

### CODE REVIEW

**Comment by elopio** [comment]
This is a branch pre-macaroons. To seed the fake servers that we will use to test parts of the store communication.

File: integration_tests/__init__.py [hunk 1]
```diff
@@ -78,13 +78,17 @@ def get_output_ignoring_non_zero_exit(self, binary, cwd):
...
         return output
 
-    def login(self, email='u1test+snapcraft@canonical.com',
+    def login(self, email=None,
               password=None, expect_success=True):
-        password = password or os.getenv('TEST_USER_PASSWORD', None)
+        email = email or os.getenv(
+            'TEST_USER_EMAIL', 'u1test+snapcraft@canonical.com')
+        password = password or os.getenv(
+            'TEST_USER_PASSWORD', 'test correct password')
```
Comment (sergiusens) [inline integration_tests/__init__.py]: shouldn't `test correct password` be an empty string? Or is the `self.skipTest` call not needed anymore?

Comment (elopio) [inline integration_tests/__init__.py]: the skip is not needed. I'll remove it.

**Comment by elopio** [comment]
retest this please

**Comment by elopio** [comment]
This is missing the fake upload server. I'm on it.

**Comment by elopio** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

**Comment by sergiusens** [comment]
retest this please

File: snapcraft/tests/fake_servers.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            'success': True
+        }
+        self.wfile.write(json.dumps(response).encode())
+
+    def do_GET(self):
+        parsed_path = urllib.parse.urlparse(self.path)
+        scan_complete_path = (
+            self._DEV_API_PATH + 'api/click-scan-complete/updown/')
+        if parsed_path.path.startswith(scan_complete_path):
+            self._handle_scan_complete_request()
```
Comment (sergiusens) [inline snapcraft/tests/fake_servers.py]: @elopio for a TODO, a counter that makes the client call this at least twice.

**Comment by sergiusens** [comment]
:+1:

**Pull request merged**